File: app/src/main/java/jp/ac/titech/itpro/sdl/gles10ex/MainActivity.java
package jp.ac.titech.itpro.sdl.gles10ex;

import android.hardware.Sensor;
import android.hardware.SensorManager;
import android.hardware.SensorEvent;
import android.hardware.SensorEventListener;
import android.hardware.SensorManager;
import android.opengl.GLSurfaceView;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.SeekBar;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity implements SeekBar.OnSeekBarChangeListener, SensorEventListener  {
    private final static String TAG = "MainActivity";

    private GLSurfaceView glView;
    private SimpleRenderer renderer;
    private SeekBar rotationBarX, rotationBarY, rotationBarZ;

    private SensorManager sensorMgr;
    private Sensor accelerometer;

    private float x = 0, y = 0, z = 0;
    private float vx, vy, vz;
    private int accuracy;

    private final static float alpha = 0.75F;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d(TAG, "onCreate");
        setContentView(R.layout.activity_main);
        glView = (GLSurfaceView) findViewById(R.id.glview);

        rotationBarX = (SeekBar) findViewById(R.id.rotation_bar_x);
        rotationBarY = (SeekBar) findViewById(R.id.rotation_bar_y);
        rotationBarZ = (SeekBar) findViewById(R.id.rotation_bar_z);
        rotationBarX.setOnSeekBarChangeListener(this);
        rotationBarY.setOnSeekBarChangeListener(this);
        rotationBarZ.setOnSeekBarChangeListener(this);

        renderer = new SimpleRenderer();
        renderer.addObj(new Cube(0.5f, 0, 0.2f, -3));
        renderer.addObj(new Pyramid(0.5f, 0, 0, 0));
        glView.setRenderer(renderer);

        sensorMgr = (SensorManager) getSystemService(SENSOR_SERVICE);
        accelerometer = sensorMgr.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
        if (accelerometer == null) {
            finish();
            return;
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "onResume");
        sensorMgr.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_FASTEST);
        glView.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause");
        glView.onPause();
        sensorMgr.unregisterListener(this);
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        if (seekBar == rotationBarX)
            renderer.setRotationX(progress);
        else if (seekBar == rotationBarY)
            renderer.setRotationY(progress);
        else if (seekBar == rotationBarZ)
            renderer.setRotationZ(progress);
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {
    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {
    }

    @Override
    public void onSensorChanged(SensorEvent event) {
        vx = alpha * vx + (1 - alpha) * event.values[0];
        vy = alpha * vy + (1 - alpha) * event.values[1];
        vz = alpha * vz + (1 - alpha) * event.values[2];
        x += vx;
        y += vy;
        z += vz;
        renderer.setRotationX(x);
        renderer.setRotationY(y);
        renderer.setRotationZ(z);
    }

    @Override
    public void onAccuracyChanged(Sensor sensor, int accuracy) {
        Log.i(TAG, "onAccuracyChanged: ");
        this.accuracy = accuracy;
    }
}

File: app/src/main/java/jp/ac/titech/itpro/sdl/gles10ex/MainActivity.kt
package jp.ac.titech.itpro.sdl.gles10ex

import android.opengl.GLSurfaceView
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.SeekBar


class MainActivity : AppCompatActivity(), SeekBar.OnSeekBarChangeListener {

    private var glView: GLSurfaceView? = null
    private var renderer: SimpleRenderer? = null
    private var rotationBarX: SeekBar? = null
    private var rotationBarY: SeekBar? = null
    private var rotationBarZ: SeekBar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "onCreate")
        setContentView(R.layout.activity_main)
        glView = findViewById(R.id.glview) as GLSurfaceView

        rotationBarX = findViewById(R.id.rotation_bar_x) as SeekBar
        rotationBarY = findViewById(R.id.rotation_bar_y) as SeekBar
        rotationBarZ = findViewById(R.id.rotation_bar_z) as SeekBar
        rotationBarX!!.setOnSeekBarChangeListener(this)
        rotationBarY!!.setOnSeekBarChangeListener(this)
        rotationBarZ!!.setOnSeekBarChangeListener(this)

        renderer = SimpleRenderer()
        renderer!!.addObj(Cube(0.5f, 0f, 0.2f, -3f))
        renderer!!.addObj(Pyramid(0.5f, 0f, 0f, 0f))
        glView!!.setRenderer(renderer)
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "onResume")
        glView!!.onResume()
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "onPause")
        glView!!.onPause()
    }

    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
        if (seekBar === rotationBarX)
            renderer!!.setRotationX(progress.toFloat())
        else if (seekBar === rotationBarY)
            renderer!!.setRotationY(progress.toFloat())
        else if (seekBar === rotationBarZ)
            renderer!!.setRotationZ(progress.toFloat())
    }

    override fun onStartTrackingTouch(seekBar: SeekBar) {}

    override fun onStopTrackingTouch(seekBar: SeekBar) {}

    companion object {
        private val TAG = "MainActivity"
    }

}
